### PR TITLE
[aws-sdk-cpp] update to 1.9.96

### LINKED ISSUE #18214: [aws-sdk-cpp] update to 1.9.x
Library name: aws-sdk-cpp

New version number: 1.9.x

Other information that may be useful (release notes, etc...)

### LINKED ISSUE #19899: [aws-sdk-cpp] Update to 1.9.96
Library name: aws-sdk-cpp

New version number: 1.9.96

Other information that may be useful (release notes, etc...)
https://github.com/aws/aws-sdk-cpp/releases

### LINKED ISSUE #19900: [aws-c-common] update to 0.6.9
Library name: aws-c-common

New version number: 0.6.9

Other information that may be useful (release notes, etc...)
https://github.com/awslabs/aws-c-common/releases

### PR DESCRIPTION
**Pull request by camarois**
Fix #18214 , #19899

Adds or updates these ports: 
s2n [1.0.17]
aws-c-auth [0.6.3]
aws-c-cal [0.5.11]
aws-c-common [0.6.9]
aws-c-compression [0.2.14]
aws-c-event-stream [0.2.7]
aws-c-http [0.6.5]
aws-c-io [0.10.7]
aws-c-mqtt [0.7.6]
aws-c-s3 [0.1.25]
aws-checksums [1.]
aws-crt-cpp [0.15.1]
aws-sdk-cpp [1.9.96]

Tested on x64-linux and x86-windows triplet. I have updated the CI baseline.

### CODE REVIEW

**Comment by xavier2k6** [comment]
1.9.96 is latest
https://github.com/microsoft/vcpkg/issues/19899

**Comment by JonLiu1993** [comment]
@camarois ,Could you please take a look:
https://dev.azure.com/vcpkg/public/_build/results?buildId=59177&view=artifacts&pathAsName=false&type=publishedArtifacts

File: ports/aws-c-auth/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    REPO awslabs/aws-c-auth
+    REF 61b6524960ad5e0c7aa2e38b343425d5941781bf # v0.6.3
+    SHA512 b5dda92e4a8796f3f1b8e2d326f57979a673f57325c921cdbc9c44273ada2f2a8eb6723f0292d223175ba4cca24508d2b635fad2af5ec7dd9e7b06db9588ede6
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:12]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/aws-c-auth/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    OPTIONS
+        -DCMAKE_MODULE_PATH=${CURRENT_INSTALLED_DIR}/share/aws-c-common # use extra cmake files
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:17]:
```suggestion
vcpkg_cmake_install()
```

File: ports/aws-c-auth/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/aws-c-auth/cmake)
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:19]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/aws-c-auth/cmake)
```

File: ports/aws-c-auth/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE 
+        ${CURRENT_PACKAGES_DIR}/bin 
+        ${CURRENT_PACKAGES_DIR}/debug/bin
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:24]:
```suggestion
        "${CURRENT_PACKAGES_DIR}/bin" 
        "${CURRENT_PACKAGES_DIR}/debug/bin"
```

File: ports/aws-c-auth/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    )
+endif()
+
+file(REMOVE_RECURSE
+	${CURRENT_PACKAGES_DIR}/debug/include
+	${CURRENT_PACKAGES_DIR}/debug/lib/aws-c-auth
+	${CURRENT_PACKAGES_DIR}/lib/aws-c-auth
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:31]:
```suggestion
	"${CURRENT_PACKAGES_DIR}/debug/include"
	"${CURRENT_PACKAGES_DIR}/debug/lib/aws-c-auth"
	"${CURRENT_PACKAGES_DIR}/lib/aws-c-auth"
```

File: ports/aws-c-auth/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	)
+
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:36]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```

File: ports/aws-c-auth/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/aws-c-auth/portfile.cmake:39]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/aws-c-auth/vcpkg.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "aws-c-auth",
+  "version": "0.6.3",
```
Comment (JonLiu1993) [inline ports/aws-c-auth/vcpkg.json:3]:
```suggestion
  "version-semver": "0.6.3",
```

File: ports/aws-c-cal/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    REPO awslabs/aws-c-cal
+    REF aa89aa4950074babe84762413f39bd364ecaf944 # v0.5.11
+    SHA512 b92ae3cb14d26dfe48cb903df56f7df91a4dc0ab2e5ea4f095c72b0b7e0a2582f1324c73eb42c080bcb0a59a3dfc37cd2912037fc8e5f7df8433899749fca63c
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:12]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/aws-c-cal/portfile.cmake [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+	OPTIONS
+        -DCMAKE_MODULE_PATH=${CURRENT_INSTALLED_DIR}/share/aws-c-common # use extra cmake files
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:17]:
```suggestion
vcpkg_cmake_install()
```

File: ports/aws-c-cal/portfile.cmake [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/aws-c-cal/cmake)
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:19]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/aws-c-cal/cmake)
```

File: ports/aws-c-cal/portfile.cmake [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+vcpkg_copy_tools(
+    TOOL_NAMES sha256_profile
+    AUTO_CLEAN
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE 
+        ${CURRENT_PACKAGES_DIR}/bin 
+        ${CURRENT_PACKAGES_DIR}/debug/bin
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:29]:
```suggestion
        "${CURRENT_PACKAGES_DIR}/bin" 
        "${CURRENT_PACKAGES_DIR}/debug/bin"
```

File: ports/aws-c-cal/portfile.cmake [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    )
+endif()
+
+file(REMOVE_RECURSE
+	${CURRENT_PACKAGES_DIR}/debug/include
+	${CURRENT_PACKAGES_DIR}/debug/lib/aws-c-cal
+	${CURRENT_PACKAGES_DIR}/lib/aws-c-cal
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:36]:
```suggestion
	"${CURRENT_PACKAGES_DIR}/debug/include"
	"${CURRENT_PACKAGES_DIR}/debug/lib/aws-c-cal"
	"${CURRENT_PACKAGES_DIR}/lib/aws-c-cal"
```

File: ports/aws-c-cal/portfile.cmake [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+	)
+
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE	${CURRENT_PACKAGES_DIR}/debug/share)
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:41]:
```suggestion
file(REMOVE_RECURSE	"${CURRENT_PACKAGES_DIR}/debug/share")
```

File: ports/aws-c-cal/portfile.cmake [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/aws-c-cal/portfile.cmake:44]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/aws-c-cal/vcpkg.json [hunk 16]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "aws-c-cal",
+  "version": "0.5.11",
```
Comment (JonLiu1993) [inline ports/aws-c-cal/vcpkg.json:3]:
```suggestion
  "version-semver": "0.5.11",
```

File: ports/aws-c-cal/vcpkg.json [hunk 17]
```diff
@@ -0,0 +1,10 @@
...
+  "description": "C99 wrapper for cryptography primitives.",
+  "homepage": "https://github.com/awslabs/aws-c-cal",
+  "supports": "!arm & !uwp",
+  "dependencies": [
+    "aws-c-common"
+  ]
```
Comment (JonLiu1993) [inline ports/aws-c-cal/vcpkg.json:9]:
```suggestion
    "aws-c-common",
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
```

File: ports/aws-c-common/vcpkg.json [hunk 18]
```diff
@@ -1,8 +1,7 @@
 {
   "name": "aws-c-common",
-  "version-string": "0.4.56",
-  "port-version": 2,
+  "version": "0.6.9",
```
Comment (JonLiu1993) [inline ports/aws-c-common/vcpkg.json:3]:
```suggestion
  "version-semver": "0.6.9",
```

File: ports/aws-c-compression/portfile.cmake [hunk 19]
```diff
@@ -0,0 +1,32 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO awslabs/aws-c-compression
+    REF 5fab8bc5ab5321d86f6d153b06062419080820ec # v0.2.14
+    SHA512 0063d0d644824d858211840115c17a33bfc2b67799e886c530ea8a42071b7bfc67bb6cf8135c538a292b8a7a6276b1d24bb7649f37ce335bc16938f2fca5cb7d
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/aws-c-compression/portfile.cmake:10]: Same as above, modify the portiflie.cmake cmake function to a new function and add "" for the path

File: ports/aws-c-compression/vcpkg.json [hunk 20]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "aws-c-compression",
+  "version": "0.2.14",
+  "description": "C99 implementation of huffman encoding/decoding",
+  "homepage": "https://github.com/awslabs/aws-c-compression",
+  "supports": "!arm & !uwp",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/aws-c-compression/vcpkg.json:7]: Same as above ,add new cmake function dependencies

File: ports/aws-c-event-stream/vcpkg.json [hunk 21]
```diff
@@ -1,10 +1,13 @@
 {
   "name": "aws-c-event-stream",
-  "version-string": "0.1.6",
-  "port-version": 1,
+  "version": "0.2.7",
```
Comment (JonLiu1993) [inline ports/aws-c-event-stream/vcpkg.json:3]:
```suggestion
  "version-semver": "0.2.7",
```

File: ports/aws-c-http/portfile.cmake [hunk 22]
```diff
@@ -0,0 +1,44 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO awslabs/aws-c-http
+    REF 6ff61d621c374ab763f94584d26b1d29baa81c2e # v0.6.5
+    SHA512 91a5f3c18e6c8d14368ad2b7275a1ebcc6398758f6b1f87ba6199998390a26e768e25c16b27f589e35371f7304b6784cfc21699ff724c811135f2f73732fe85b
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/aws-c-http/portfile.cmake:10]: Same as above, modify the portiflie.cmake cmake function to a new function and add "" for the path

File: ports/aws-c-http/vcpkg.json [hunk 23]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "aws-c-http",
+  "version": "0.6.5",
+  "description": "C99 implementation of the HTTP/1.1 and HTTP/2 specifications",
+  "homepage": "https://github.com/awslabs/aws-c-http",
+  "supports": "!arm & !uwp",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/aws-c-http/vcpkg.json:7]: Same as above

File: ports/aws-c-io/portfile.cmake [hunk 24]
```diff
@@ -0,0 +1,32 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO awslabs/aws-c-io
+    REF 57b00febac48e78f8bf8cff4c82a249e6648842a # v0.10.7
+    SHA512 ffcf5ba4ccdff23ca390fc4eb935f88040447589886348234aa1c24b531401521df99a6ac578c1679a3c1a06dfcef6deb833a0b9d53e31d42576a3ad03ade6fc
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/aws-c-io/portfile.cmake:10]: Same as above

File: ports/aws-c-io/vcpkg.json [hunk 25]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "aws-c-io",
+  "version": "0.10.7",
```
Comment (JonLiu1993) [inline ports/aws-c-io/vcpkg.json:3]:
```suggestion
  "version-semver": "0.10.7",
```

File: ports/aws-c-io/vcpkg.json [hunk 26]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "Handles all IO and TLS work for application protocols.",
+  "homepage": "https://github.com/awslabs/aws-c-io",
+  "supports": "!arm & !uwp",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/aws-c-io/vcpkg.json:7]: Same as above

File: ports/aws-c-mqtt/portfile.cmake [hunk 27]
```diff
@@ -0,0 +1,44 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO awslabs/aws-c-mqtt
+    REF 0a70bf814845e487b7e4862af7ad9e4a1199b5f4 # v0.7.6
+    SHA512 9c8719004e9e4c56f270c5c5dcb41b5ecacb050dadf2a548ab2119a1a8f223c9117829346e2e86c8c77f6efe3502019085344fe4f470ccdd646dc87a3d4f4c3f
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/aws-c-mqtt/portfile.cmake:10]: Save as above

File: ports/aws-c-mqtt/vcpkg.json [hunk 28]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "aws-c-mqtt",
+  "version": "0.7.6",
+  "description": "C99 implementation of the MQTT 3.1.1 specification.",
+  "homepage": "https://github.com/awslabs/aws-c-mqtt",
+  "supports": "!arm & !uwp",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/aws-c-mqtt/vcpkg.json:7]: Save as above

File: ports/aws-c-s3/portfile.cmake [hunk 29]
```diff
@@ -0,0 +1,39 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO awslabs/aws-c-s3
+    REF e38bdad12bd4c618e8d10489374d87e6ae3f7a19 # v0.1.25
+    SHA512 1ed3313419416ce28373968cb692cf67230b1efced4e187cd5047b146db3bc769dc283dff1af2e41b05ad6f50cbeea58718a34945a93645abacca2515f2a6cf1
+    HEAD_REF master
+    PATCHES fix-cmake-target-path.patch
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/aws-c-s3/portfile.cmake:10]: Save as above

File: ports/aws-c-s3/vcpkg.json [hunk 30]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "aws-c-s3",
+  "version": "0.1.25",
```
Comment (JonLiu1993) [inline ports/aws-c-s3/vcpkg.json:3]:
```suggestion
  "version-semver": "0.1.25",
```

File: ports/aws-c-s3/vcpkg.json [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "C99 library implementation for communicating with the S3 service, designed for maximizing throughput on high bandwidth EC2 instances.",
+  "homepage": "https://github.com/awslabs/aws-c-s3",
+  "supports": "!arm & !uwp",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/aws-c-s3/vcpkg.json:7]: Save as above

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
>Adds or updates these ports:
>s2n [1.0.17]

That port failed to build on MacOS. https://dev.azure.com/vcpkg/public/_build/results?buildId=59340

**Comment by camarois** [comment]
@BillyONeal I fixed the failing ports.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Looks good, thanks :)

**Pull request merged**